### ISSUE TITLE
Investigate into null values in startup metrics

### ISSUE THREAD

**Issue by rocketsroger** [mozilla-mobile/fenix#19147]
Currently there are large number of startup metrics are reported as null.  Investigate into how startup metrics are reported to understand the cause/solution of the issue.

Example query: https://sql.telemetry.mozilla.org/queries/79505/

**Comment by wlach** [comment]
@rocketsroger when you've diagnosed this issue and have a fix, could you also provide a list of affected metrics? This would be a handy thing to have in the commentary section in the glean dictionary so people looking at the data are aware of this issue / gotcha. See the commentary section here, for example: https://dictionary.protosaur.dev/apps/firefox_ios/metrics/app_opened_as_default_browser

**Comment by rocketsroger** [comment]
Glean team understand the current limitation and has opened an issue for this as well.  https://bugzilla.mozilla.org/show_bug.cgi?id=1706637

**Comment by rocketsroger** [comment]
As a workaround suggestion in https://bugzilla.mozilla.org/show_bug.cgi?id=1706637.  I will move any startup metrics operations to right after Glean initialization.  I will create another issue to track the follow up.

**Comment by rocketsroger** [comment]
follow up issue created: https://github.com/mozilla-mobile/fenix/issues/19251

**Comment by mcomella** [comment]
I analyzed the perf impact of this and it's very large: 114ms or -8.07% in time to visual completeness on the Moto G5. This is a very large regression so I don't think we should keep this change and have recommendations of alternative implementations: https://github.com/mozilla-mobile/fenix/issues/19395#issuecomment-832873039

**Comment by rocketsroger** [comment]
> I analyzed the perf impact of this and it's very large: 114ms or -8.07% in time to visual completeness on the Moto G5. This is a very large regression so I don't think we should keep this change and have recommendations of alternative implementations: [#19395 (comment)](https://github.com/mozilla-mobile/fenix/issues/19395#issuecomment-832873039)

Agreed.  I was expecting performance degradation but I didn't think it was this much.  Since this is a workaround until Glean team's investigation.  I will create a PR to move the metrics setup off the main thread.  If moving metrics setup off the main thread causes the workaround to not work consistently then there's nothing we can do.

**Comment by rocketsroger** [comment]
Looking at the query we have about 20% null values still.  This is probably the best we can do without affecting performance.  Once Glean team's investigation is done I will use the follow up issue https://github.com/mozilla-mobile/fenix/issues/19251 to fix the root cause.  Closing this as done.
